app/controllers/posts_controller.rb

```diff
@@ -1,13 +1,40 @@
 class PostsController < ApplicationController
   def index
+    @posts = Post.all
   end
 
   def show
+    @post = Post.find(params[:id])
   end
 
   def new
+    @post = Post.new
   end
 
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :body))
+    @post.user = current_user
+    if @post.save
+      flash[:notice] = "Post was successfully saved."
+      redirect_to @post
+    else
+      flash[:error] = "There was an error saving the post. Please try again."
+      render :new
+    end
+  end  
+
   def edit
+    @post = Post.find(params[:id])
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    if @post.update_attributes(params.require(:post).permit(:title, :body))
```

Comment:
Consider extracting a private `post_params` method to DRY up the `params.require(:post).permit(:title, :body)` statements on line 15 and line 32.